### ISSUE TITLE
lines_filter_map_ok should respect MSRV

### ISSUE THREAD

**Issue by taiki-e** [rust-lang/rust-clippy#14127]
### Summary

`Iterator::map_while` requires Rust 1.57, but this lint also suggests using it for code with MSRV less than 1.57.

### Lint Name

lines_filter_map_ok

### Reproducer

I tried this code:

```rust
#![warn(clippy::lines_filter_map_ok)]

use std::{fs::File, io::*};

#[clippy::msrv = "1.56"]
fn _f() {
    let _lines = BufReader::new(File::open("some-path").unwrap())
        .lines()
        .filter_map(Result::ok);
}
```

I saw this happen:

```
warning: `filter_map()` will run forever if the iterator repeatedly produces an `Err`
 --> src/lib.rs:9:10
  |
9 |         .filter_map(Result::ok);
  |          ^^^^^^^^^^^^^^^^^^^^^^ help: replace with: `map_while(Result::ok)`
  |
note: this expression returning a `std::io::Lines` may produce an infinite number of `Err` in case of a read error
 --> src/lib.rs:7:18
  |
7 |       let _lines = BufReader::new(File::open("some-path").unwrap())
  |  __________________^
8 | |         .lines()
  | |________________^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#lines_filter_map_ok
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(clippy::lines_filter_map_ok)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

However, the suggested code cases compile error on MSRV (Rust 1.56): https://godbolt.org/z/4KT7M4WTr

```
error[E0658]: use of unstable library feature 'iter_map_while': recently added
 --> <source>:9:10
  |
9 |         .map_while(Result::ok);
  |          ^^^^^^^^^
  |
  = note: see issue #68537 <https://github.com/rust-lang/rust/issues/68537> for more information
```

I expected to see this happen: no warning or another suggestion

[playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=a7cbbb9251eb10a764c2adc4e1131d6c)

### Version

```text
rustc 1.86.0-nightly (854f22563 2025-01-31)
binary: rustc
commit-hash: 854f22563c8daf92709fae18ee6aed52953835cd
commit-date: 2025-01-31
host: aarch64-apple-darwin
release: 1.86.0-nightly
LLVM version: 19.1.7
```
